### PR TITLE
refactored test_resources.py into test_resources package

### PR DESCRIPTION
**Pull request by bvedang**
**Problem**

What problem have you solved?
The test_resources.py file had become large and cluttered with numerous classes, making it difficult to navigate, maintain, and understand the scope of individual tests. This complexity could potentially hinder future development and decrease overall code quality.

**Solution**
To address the issue, I refactored the test_resources.py file by organizing its contents into a more structured package named test_resources. This involved logically grouping related classes into separate module files within the package.

How did you solve the problem?
This involved logically grouping related classes into separate module files within the package.

The new structure is as follows:

**`basic_resources.py`**: Contains foundational resource classes.
**`enhanced_resources.py`**: Contains resource classes with enhanced features.
**`widgets.py`**: Hosts widget-related classes.
**`widget_tests.py`**: Contains tests specifically for widgets.
... [and so on for other modules]

**Acceptance Criteria**

Have you written tests? Have you included screenshots of your changes if applicable?
Did you document your changes? 
No new tests were written as this was a refactoring task. But i made sure to run the make test command and check if all the test are passing

### CODE REVIEW

File: .DS_Store
Comment (matthewhegarty) [inline .DS_Store:1]: Please can you remove this file from the PR.  Also it would be useful to update .gitignore so that this is ignored.

**Review by matthewhegarty — CHANGES_REQUESTED** [review]
Thanks for working on this.

When I run `make test` in `release-4` I get 475 tests run.
In this branch, I get 306, so some tests are being missed.  This is probably because some of the new modules with tests in are not named 'test_*'. If you rename these modules it should work.

Feel free to update the changelog, and also add your name to AUTHORS if you like

File: tests/core/tests/test_resources/basic_resources.py [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+from unittest.mock import patch
+
+import tablib
+from django.test import TestCase
+
+from import_export import fields, resources, results
+from tests.core.models import Author, Book, Category, Profile, WithDefault
+
+
+class MyResource(resources.Resource):
```
Comment (matthewhegarty) [inline tests/core/tests/test_resources/basic_resources.py:12]: I suggest take all Resource classes out of test modules and put them in a new module called 'resources`, that are then imported into test modules.  This way we separate tests and helper classes.

**Comment by matthewhegarty** [comment]
There is also a test failure due to a deprecation added to django.  If you merge `release-4` it should correct it

File: tests/core/tests/test_resources/basic_resources.py (hunk 1 above)
Comment (bvedang) [reply tests/core/tests/test_resources/basic_resources.py:12]: Just want to clarify if we want to keep the resources module in folder `tests/core` or totally outside of parent `tests` folder.

File: .DS_Store
Comment (bvedang) [reply .DS_Store:1]: Sure i will get it done.

File: tests/core/tests/test_resources/basic_resources.py (hunk 1 above)
Comment (matthewhegarty) [reply tests/core/tests/test_resources/basic_resources.py:12]: The resources in here should apply to tests only, so I suggest to move to a module in 

```
tests/core/tests
```

File: tests/core/tests/test_resources/basic_resources.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+from collections import OrderedDict
+from unittest import mock
+from unittest.mock import patch
+
+import tablib
+from django.test import TestCase
+
+from import_export import fields, resources, results
+from tests.core.models import Author, Book, Category, Profile, WithDefault
```
Comment (matthewhegarty) [inline tests/core/tests/test_resources/basic_resources.py:9]: Please can you make the import from `core`, so this should be:

```
from core.models import Author, Book, Category, Profile, WithDefault
```

**Comment by bvedang** [comment]
I have implemented the recommended changes from the review. The resources have been successfully extracted and restructured into tests. I've verified that the total number of tests remains consistent at 475, and all tests exhibit the same pass rate as before the refactor.

**Comment by matthewhegarty** [comment]
Thanks.  Am seeing one test failure in `test_resources.py`, the target string needs to be changed to:

```
"Field x does not exists in <class 'core.tests.test_resources.MyResource'> resource"
```

**Review by matthewhegarty — APPROVED** [review]
That's great.  Thank you Vedang, much appreciated.  Will merge.

**Comment by bvedang** [comment]
Hi Matthew,
I'm relatively new to the open-source community, and I just made my first contribution here.I'd love to get your feedback on how I could improve my contribution. Could you spare a few moments to take a look and suggest any enhancements?

Thank you for your guidance!

**Comment by matthewhegarty** [comment]
Sure - no problem - I'll email you.

**Pull request merged**